src/Plugin/Block/RecentGroupContentBlock.php

```diff
@@ -0,0 +1,265 @@
+    if (empty($this->ogContext->getGroup())) {
+      return [];
+    }
+
+    $list = array_map(function ($entity) {
+      return [
+        '#type' => 'link',
+        '#title' => $entity->label(),
+        '#url' => $entity->toUrl(),
+      ];
+    }, $this->getGroupContent());
+
+    return [
+      'list' => [
+        '#theme' => 'item_list',
+        '#items' => $list,
+      ],
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCacheTags() {
+    $tags = parent::getCacheTags();
+    if ($group = $this->ogContext->getGroup()) {
+      $tag = $group->getEntityTypeId() . ':' . $group->id();
+      $tags = Cache::mergeTags(Cache::buildTags('og-group-content', [$tag]), $tags);
+    }
+    return $tags;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCacheContexts() {
+    // The block varies by user because of the access check on the query that
+    // retrieves the group content.
+    return Cache::mergeContexts(parent::getCacheContexts(), ['user', 'og_group_context']);
+  }
+
+  /**
+   * Returns the most recent group content for the active group.
+   *
+   * @return \Drupal\Core\Entity\EntityInterface[]
+   *   The most recent group content for the group which is currently active
+   *   according to OgContext.
+   */
+  protected function getGroupContent() {
+    $group = $this->ogContext->getGroup();
+    $entity_type = $this->configuration['entity_type'];
+    $bundles = array_filter($this->configuration['bundles'][$entity_type]);
+    $definition = $this->entityTypeManager->getDefinition($entity_type);
+
+    // Retrieve the fields which reference our entity type and bundle.
+    $query = $this->entityTypeManager
+      ->getStorage('field_storage_config')
+      ->getQuery()
+      ->condition('type', OgGroupAudienceHelperInterface::GROUP_REFERENCE)
+      ->condition('entity_type', $entity_type);
+
+    /** @var \Drupal\field\FieldStorageConfigInterface[] $fields */
+    $fields = array_filter(FieldStorageConfig::loadMultiple($query->execute()), function (FieldStorageConfigInterface $field) use ($group) {
+      $type_matches = $field->getSetting('target_type') === $group->getEntityTypeId();
+      // If the list of target bundles is empty, it targets all bundles.
+      $bundle_matches = empty($field->getSetting('target_bundles')) || in_array($group->bundle(), $field->getSetting('target_bundles'));
+      return $type_matches && $bundle_matches;
+    });
+
+    // Compile the group content.
+    $ids = [];
+    foreach ($fields as $field) {
+      // Query all group content that references the group through this field.
+      $results = $this->entityTypeManager
+        ->getStorage($entity_type)
+        ->getQuery()
+        ->condition($field->getName() . '.target_id', $group->id())
+        ->condition($definition->getKey('bundle'), $bundles, 'IN')
+        ->accessCheck()
+        // @todo Add support for entity types that use a different column name
```

Comment:
No I think this is actually really hard to do. In core we have only a few entity keys that we can depend on, and 'created' is not one of them. We don't even have an interface for the created date that we can use to detect if an entity type even supports a creation timestamp. The closest we have is `EntityChangedInterface`.

To fix this we probably need to introduce an 'official' entity key for the creation date in core. This `@todo` is mainly here to inform people that we know this problem exists :)

I think the most practical way to circumvent this for a custom entity type is to simply subclass the block and override the method.